### DISCUSSION TITLE
Getter called five times on each proxy object update

### DISCUSSION THREAD

**Discussion by crabvk** [pmndrs/valtio#872] [Bug report] [answered]
Hello.
I've noticed that for each proxy object update the snapshot getter of that object called five times.
Here is an example application to demonstrate the issue https://github.com/crabvk/valtio-getter. As you can see in the browser console with each press of the button there're five "Value get" log entries.
Is this a bug or it works as intended? If the latter please explain why the getter called five times with each update.

**Comment by dai-shi** [comment, marked as answer]
class getters are handled as just function call on the prototype chain.
Valtio doesn't cache prototype function call.
On the other hand, object getters are cached. Here's the modified example: https://stackblitz.com/edit/github-utfgww

Though, five times sounds a bit too many. Not sure how it could be improved.

**Comment by crabvk** [comment]
Thank you for quick response and example, gonna use plain objects instead of classes then.
